File: sdk/clojure/src/hitechcloud/sdk/ipam.clj
(ns hitechcloud.sdk.ipam
  (:require [hitechcloud.sdk.http-client :as http]))

;; HiTechCloudIPAM: IP addresses, subnets, reverse DNS

(defn list_ips "GET /api/service/{id}/htcipam/ips - List IP addresses" [client & [params]] (http-get client "/api/service/{id}/htcipam/ips" params))

(defn list_subnets "GET /api/service/{id}/htcipam/subnets - List subnets" [client & [params]] (http-get client "/api/service/{id}/htcipam/subnets" params))

(defn list_rdns "GET /api/service/{id}/htcipam/rdns - List reverse DNS records" [client & [params]] (http-get client "/api/service/{id}/htcipam/rdns" params))

(defn set_rdns "POST /api/service/{id}/htcipam/rdns - Set reverse DNS" [client & [params]] (http-post client "/api/service/{id}/htcipam/rdns" params))

File: sdk/fsharp/src/HiTechCloudSDK/Ipam.fs
namespace HiTechCloud.SDK

/// HiTechCloudIPAM: IP addresses, subnets, reverse DNS
type IpamResource(httpClient: HttpClient) =

    /// GET /api/service/{id}/htcipam/ips - List IP addresses
    member this.ListIps() =
        async {
            return! httpClient.getAsync("/api/service/{id}/htcipam/ips")
        }

    /// GET /api/service/{id}/htcipam/subnets - List subnets
    member this.ListSubnets() =
        async {
            return! httpClient.getAsync("/api/service/{id}/htcipam/subnets")
        }

    /// GET /api/service/{id}/htcipam/rdns - List reverse DNS records
    member this.ListRdns() =
        async {
            return! httpClient.getAsync("/api/service/{id}/htcipam/rdns")
        }

    /// POST /api/service/{id}/htcipam/rdns - Set reverse DNS
    member this.SetRdns() =
        async {
            return! httpClient.postAsync("/api/service/{id}/htcipam/rdns")
        }

